### PR TITLE
Fixed/Implemented compilation bugs, Arth router pool functions

### PR DESCRIPTION
**Pull request by Sagar133**

### CODE REVIEW

File: contracts/Staking/StakingRewards.sol [hunk 1]
```diff
@@ -265,6 +265,19 @@ contract StakingRewards is
 
     /* ========== MUTATIVE FUNCTIONS ========== */
 
+    function receiveMint(
```
Comment (senamakel) [inline contracts/Staking/StakingRewards.sol]: do we need this anymore? we have a stakeFor function now

File: contracts/Staking/IMintAndCallFallBack.sol [hunk 2]
```diff
@@ -6,6 +6,7 @@ interface IMintAndCallFallBack {
     function receiveMint(
```
Comment (senamakel) [inline contracts/Staking/IMintAndCallFallBack.sol]: probably remove this?

File: contracts/ERC20/Variants/AnyswapV4Token.sol [hunk 3]
```diff
@@ -147,7 +147,7 @@ contract AnyswapV4Token is ERC20Custom, AccessControl, IAnyswapV4Token {
                 verifyPersonalSign(target, hashStruct, v, r, s)
         );
 
-        _approve(owner, spender, value);
+        _approve(target, spender, value);
```
Comment (senamakel) [inline contracts/ERC20/Variants/AnyswapV4Token.sol:208]: how is this possible? i thought we copy pasted the code?

File: contracts/Staking/StakingRewards.sol (hunk 1 above)
Comment (ghost) [reply contracts/Staking/StakingRewards.sol]: Don't need it.

File: contracts/Staking/IMintAndCallFallBack.sol (hunk 2 above)
Comment (ghost) [reply contracts/Staking/IMintAndCallFallBack.sol]: Yes, will remove it.

File: contracts/ERC20/Variants/AnyswapV4Token.sol (hunk 3 above)
Comment (ghost) [reply contracts/ERC20/Variants/AnyswapV4Token.sol:208]: Yes, it's copy pasted however there's no declaration of owner variable in that function even in the reference code.

**Pull request merged**